jupyter_server/gateway/connections.py

```diff
@@ -147,6 +147,9 @@ def handle_incoming_message(self, message: str) -> None:
         """Send message to gateway server."""
         if self.ws is None and self.ws_future is not None:
             loop = IOLoop.current()
+            if self.ws_future.done():
+                self.log.error(f"Exception connect to  gateway server {self.ws_future.exception()}")
```

Comment:
`future.done()` does not imply that it failed, whereas log message assumes it has.

Maybe the fix here is to add `and not self.ws_future.done()` to the condition on line 148? Or more lkely, the ws future error was already logged, and the right thing to do here is a debug-log for "message on closed websocket"?

Reply:
Yeah，you are right，I adjust it

Reply:
maybe like this 
```
            if self.ws_future.done() and isinstance(self.ws_future.exception(), Exception):
                self.log.error(f"Exception connect to server {self.ws_future.exception()}")
                return
```
if use `if self.ws is None and not self.ws_future.done()` when raise connection err，code will execute else logic `self._write_message(message)`